src/v/datalake/schema_parquet.cc

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+
+#include "iceberg/datatypes.h"
+#include "serde/parquet/schema.h"
+
+namespace datalake {
+namespace {
+
+struct primitive_type_converting_visitor {
+    serde::parquet::schema_element operator()(const iceberg::boolean_type&) {
+        return serde::parquet::schema_element{
+          .type = serde::parquet::bool_type{},
+        };
+    }
+    serde::parquet::schema_element operator()(const iceberg::int_type&) {
+        return serde::parquet::schema_element{
+          .type = serde::parquet::i32_type{},
+          .logical_type
+          = serde::parquet::int_type{.bit_width = 32, .is_signed = true},
+        };
+    }
+    serde::parquet::schema_element operator()(const iceberg::long_type&) {
+        return serde::parquet::schema_element{
+          .type = serde::parquet::i64_type{},
+          .logical_type
+          = serde::parquet::int_type{.bit_width = 64, .is_signed = true},
+        };
+    }
+    serde::parquet::schema_element operator()(const iceberg::float_type&) {
+        return serde::parquet::schema_element{
+          .type = serde::parquet::f32_type{},
+        };
+    }
+    serde::parquet::schema_element operator()(const iceberg::double_type&) {
+        return serde::parquet::schema_element{
+          .type = serde::parquet::f64_type{},
+        };
+    }
+    serde::parquet::schema_element
+    operator()(const iceberg::decimal_type& type) {
+        return serde::parquet::schema_element{
+            .type = serde::parquet::byte_array_type{},
```

Comment:
should this have a fixed size of 16?

Reply:
i think it can, we are using `int128` to represent decimals